### DISCUSSION TITLE
scroll 위치 측정 시 미세한 값 차이 발생

### DISCUSSION THREAD

**Discussion by JiyeonP** [wafflestudio/seminar-2021#639] [Q&A] [answered]
과제4의 페이지네이션 구현 도중 scroll이 끝까지 내려갔는지 구현하는 부분에서 문제가 발생하여 질문 드립니다.
useRef를 사용하여 comments ul에 current를 설정한 상태입니다.
스크롤을 최대한 내렸을때도 새 페이지를 불러오지 않아서 scrollTop, scrollHeight, clientHeight을 출력해보았더니 아래의 사진과 같이 근소한 차이로 scrollTop + clientHeight >= scrollHeight이 성립하지 않는 것을 확인했습니다. 맨 아래쪽 값이 스크롤을 최대한 내렸을 때의 값입니다.
왜 이렇게 값이 측정되고 있는 것인지 그 원인을 잘 모르겠는데, 도움 주실 수 있으실까요?

![image](https://user-images.githubusercontent.com/66158942/140517230-9620f9ab-fcdb-4108-b73e-5a24eaa11a7f.png)

**Comment by woohm402** [comment, marked as answer]
1. scrollTop은 항상 integer value일 텐데, 혹시 어떤 브라우저에서 구현 중이시고 어떤 값을 출력하셨나요?
2. 꼭 그거 때문이 아니더라도 항상 `scrollTop + clientHeight >= scrollHeight` 이렇게 스크롤 끝에 도달했을 때를 정확하게 체크하지는 않습니다. 좋은 사용자 경험을 위해 `scrollTop + clientHeight >= scrollHeight - 500` 이렇게 충분한 n픽셀 전에 다음 페이지를 로드하는 것도 좋습니다.

**Reply by JiyeonP** [reply]
크롬에서 구현중이고, ul에 할당한 ref의 current.scrollHeight, current.scrollTop, current.clientHeight를 출력했습니다.

**Reply by woohm402** [reply]
[![image](https://user-images.githubusercontent.com/39977696/140523393-3895513c-91d4-44f0-bff2-8550e0a46377.png)](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop)

아하..! 처음 알았네요 감사합니다
2번 방식 차용해주시는 게 좋을 것 같아요

**Reply by JiyeonP** [reply]
넵! 감사합니다
